staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store_test.go

```diff
@@ -2298,6 +2299,60 @@ func TestStoreWatch(t *testing.T) {
 	}
 }
 
+// regression test for #108414
```

Comment:
@wojtek-t this covers a bit more than the unit tests I've added before

Reply:
You're effectively going into details of cacher in the test (e.g. casting to it, stopping it, etc.).

Can you move this test down the stack, and make it a test of cacher (under pkg/storage/cacher)?
This feels like a better place for this test.

Reply:
I'm sorry - I can't see that. Can you point me to it?
[I still don't think this is the right place for having such test]

Reply:
pinged you in a new comment, will remove this test. I've added it in a new commit because I already had doubts :smile: